modules/ingest-common/src/main/java/org/opensearch/ingest/common/CommunityIDProcessor.java

```diff
@@ -0,0 +1,654 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.ingest.common;
+
+import org.opensearch.common.hash.MessageDigests;
+import org.opensearch.common.network.InetAddresses;
+import org.opensearch.core.common.Strings;
+import org.opensearch.ingest.AbstractProcessor;
+import org.opensearch.ingest.ConfigurationUtils;
+import org.opensearch.ingest.IngestDocument;
+import org.opensearch.ingest.Processor;
+
+import java.security.MessageDigest;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.opensearch.ingest.ConfigurationUtils.newConfigurationException;
+
+/**
+ * Processor that generating community id flow hash for the network flow tuples, the algorithm is defined in
+ * <a href="https://github.com/corelight/community-id-spec">Community ID Flow Hashing</a>.
+ */
+public class CommunityIDProcessor extends AbstractProcessor {
```

Comment:
Apologies, missed that before, could you please use the conventional name (we use HTML -> Html, ID -> Id, ...), thank you:

```suggestion
public class CommunityIdProcessor extends AbstractProcessor {
```